state/validation_test.go

```diff
@@ -282,22 +284,26 @@ func TestValidateFailBlockOnCommittedEvidence(t *testing.T) {
 func TestValidateAlreadyPendingEvidence(t *testing.T) {
 	var height int64 = 1
 	state, stateDB, _ := makeState(2, int(height))
+	addr, _ := state.Validators.GetByIndex(0)
+	addr2, _ := state.Validators.GetByIndex(0)
+	ev := types.NewMockEvidence(height, time.Now(), addr)
+	ev2 := types.NewMockEvidence(height, time.Now(), addr2)
+
+	evpool := &mocks.EvidencePool{}
+	evpool.On("IsPending", mock.AnythingOfType("types.MockEvidence")).Return(func(e types.Evidence) bool {
+		return e.Equal(ev)
+	})
```

Comment:
A perhaps clearer way to write this would be:

```
evpool.On("IsPending", ev).Return(true)
evpool.On("IsPending", ev2).Return(false)
```

Reply:
ahhh I didn't know you could do that - yes that would be better